frontend/src/app/user.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@angular/core';
+import { Headers, Http, Response, RequestOptions } from '@angular/http';
+import { Observable } from 'rxjs/Rx';
+
+@Injectable()
+export class UserService {
+  private url = 'http://localhost:8000/api/login';
+
+  constructor (
+    private http: Http
+  ) {}
+
+  loginUser(username: string, password: string) {
+    let headers: Headers = new Headers({
+      'Content-Type': 'application/json',
+      'Access-Control-Allow-Origin': '*'
+    });
+    let options: RequestOptions = new RequestOptions({ headers: headers });
+
+    return this.http.post(this.url, { username, password }, options)
+    .map((res:Response) => res.json())
+    .catch(this.handleError);
+  }
+
+  handleError(reject: Response | any) {
```

Comment:
`private` please

there is not so much sense to have `Response | any` as type definition